File: DAY4/성욱/에디터 - 1406.js
const input = require('fs')
  .readFileSync('/dev/stdin')
  .toString()
  .trim()
  .split('\n')

const str = input[0]
const m = parseInt(input[1])

const left = [...str]
const right = []

for (let i = 2; i < 2 + m; i++) {
  const cmd = input[i].split(' ')

  if (cmd[0] === 'L') {
    if (left.length > 0) {
      right.push(left.pop())
    }
  } else if (cmd[0] === 'D') {
    if (right.length > 0) {
      left.push(right.pop())
    }
  } else if (cmd[0] === 'B') {
    if (left.length > 0) {
      left.pop()
    }
  } else if (cmd[0] === 'P') {
    left.push(cmd[1])
  }
}

console.log(left.join('') + right.reverse().join(''))

File: DAY4/원정/에디터_1406.py
import sys
input = sys.stdin.readline

left = list(input().strip())
right = []

for _ in range(int(input())):
    command = list(input().split())
    action = command[0]
    if action == 'L':
        if left:
            right.append(left.pop())
    elif action == 'D':
        if right:
            left.append(right.pop())
    elif action == 'B':
        if left:
            left.pop()
    else:
        left.append(command[1])

left.extend(reversed(right))
print(''.join(left))